script.js

```diff
@@ -2,71 +2,78 @@ if (localStorage.getItem('data') === null) {
-  const books = document.getElementById('books');
-  const liToRemove = document.querySelectorAll('li');
-  liToRemove.forEach((item) => {
-    item.remove();
-  });
+  book(title, author) {
+    this.bookTemplate.id = Date.now();
+    this.bookTemplate.title = title;
+    this.bookTemplate.author = author;
+    return this.bookTemplate;
+  }
 
-  bookList.forEach((book) => {
-    const li = document.createElement('li');
-    const hr = document.createElement('hr');
-    const removeBtn = document.createElement('input');
-    removeBtn.id = book.id;
-    removeBtn.className = 'remove-btn';
-    removeBtn.setAttribute('onclick', `removeBook( ${book.id} )`);
-    removeBtn.setAttribute('type', 'button');
-    removeBtn.setAttribute('value', 'Remove');
-    const titleBlock = document.createElement('div');
-    const title = document.createTextNode(book.title);
-    titleBlock.appendChild(title);
-    const authorBlock = document.createElement('div');
-    const author = document.createTextNode(book.author);
-    authorBlock.appendChild(author);
+  add(title, author) {
+    this.books.push(this.book(title, author));
+    this.save();
+  }
 
-    li.appendChild(titleBlock);
-    li.appendChild(authorBlock);
-    li.appendChild(removeBtn);
-    li.appendChild(hr);
-    books.appendChild(li);
-  });
-}
+  remove(id) {
+    this.books = this.books.filter((book) => book.id !== Number(id));
+    this.save();
+  }
 
-function getFromLocalStorage() {
-  return JSON.parse(localStorage.getItem('data'));
-}
+  save() {
+    localStorage.setItem('data', JSON.stringify(this.books));
+    this.updateView();
+  }
+
+  retrieve() {
+    this.books = JSON.parse(localStorage.getItem('data'));
+  }
 
-function saveToLocalStorage(bookList) {
-  localStorage.setItem('data', JSON.stringify(bookList));
-  addBooks(getFromLocalStorage());
+  updateView() {
+    const books = document.getElementById('books');
+    const liToRemove = document.querySelectorAll('li');
+    liToRemove.forEach((item) => {
+      item.remove();
+    });
+    this.retrieve();
+    this.books.forEach((book) => {
+      const li = document.createElement('li');
+      const hr = document.createElement('hr');
+      const removeBtn = document.createElement('input');
+      removeBtn.id = book.id;
+      removeBtn.className = 'remove-btn';
+      removeBtn.addEventListener('click', (ev) => {
+        this.remove(ev.target.id);
+      });
+      removeBtn.setAttribute('type', 'button');
+      removeBtn.setAttribute('value', 'Remove');
+      const titleBlock = document.createElement('div');
+      const title = document.createTextNode(book.title);
+      titleBlock.appendChild(title);
+      const authorBlock = document.createElement('div');
+      const author = document.createTextNode(book.author);
+      authorBlock.appendChild(author);
+      li.appendChild(titleBlock);
+      li.appendChild(authorBlock);
+      li.appendChild(removeBtn);
+      li.appendChild(hr);
+      books.appendChild(li);
```

Comment:
- [ ] Please try to use DOM manipulation outside of class. Classes are a template for creating objects. 

![image](https://user-images.githubusercontent.com/53019916/124639677-10a53080-deaa-11eb-9a36-5377b0a96c4a.png)